concurrent_test.go

```diff
@@ -106,34 +116,56 @@ func concurrentReadAndWrite(t *testing.T,
 	minWriteBytes, maxWriteBytes int,
 	testDuration time.Duration) {
 
-	// prepare the db
+	t.Log("Preparing db.")
 	db := btesting.MustCreateDB(t)
 	err := db.Update(func(tx *bolt.Tx) error {
 		_, err := tx.CreateBucket(bucket)
 		return err
 	})
 	require.NoError(t, err)
 
+	t.Log("Starting workers.")
+	records := runWorkers(t,
+		db, bucket, keys,
+		readerCount, minReadInterval, maxReadInterval,
+		minWriteInterval, maxWriteInterval, minWriteBytes, maxWriteBytes,
+		testDuration)
+
+	t.Log("Analyzing the history records.")
+	if err := analyzeHistoryRecords(records); err != nil {
+		t.Errorf("The history records are not linearizable:\n %v", err)
+	}
+
+	saveDataIfFailed(t, db, records)
```

Comment:
defer it to ensure it is executed even on fatal?

Reply:
If somehow panicking during test, then the records hasn't been collected yet. I may need change big change. Let me think about separately. thx